### DISCUSSION TITLE
How can I update git?

### DISCUSSION THREAD

**Discussion by devercises** [microsoft/WSL#6735] [Q&A] [answered]
I have 2.25 version of git and I want to update to 2.31, which is newest version. How can I do this??

My distro: Ubuntu 20.04.2 LTS
Wsl version: 2

**Comment by DHowett** [comment, marked as answer]
It is no different from upgrading git on Ubuntu _without_ WSL.

**Comment by therealkenc** [comment]
```
$ sudo add-apt-repository ppa:git-core/ppa
$ sudo apt update && sudo apt upgrade
```

**Reply by chankruze** [reply]
This should be the correct answer.

**Reply by KangYeonCheol** [reply]
If you find the error message such as sudo: add-apt-repository: command not found,
then you have to install software-properties-common as follow:

sudo apt-get install software-properties-common

after that,

sudo apt-get update

**Comment by tlunsfordCXP** [comment]
Copying the previous answer witout the leading $, for easy copying:

```
sudo add-apt-repository ppa:git-core/ppa
sudo apt update && sudo apt upgrade
```

**Comment by goluvr** [comment]
You probably also need `sudo apt install git`. <br>The following three lines worked for me (per https://therdnotes.com/update-git-in-wsl2-ubuntu and https://qiita.com/takaha4k/items/c0ce8820990f715ed4b9): 
```
sudo add-apt-repository ppa:git-core/ppa  
sudo apt update  
sudo apt install git
```
